interop-tests/src/lib.rs

```diff
@@ -0,0 +1,319 @@
+                target_data,
+                &[HashAlgorithm::Sha256],
+            )
+            .unwrap();
+    }
+    let step_str = format!("{}", step);
+    let target_data = step_str.as_bytes();
+
+    let signed_targets = targets_builder
+        .signed::<JsonPretty>(keys.get("targets").unwrap())
+        .unwrap();
+    let targets = signed_targets.assume_valid().unwrap();
+
+    let hash = targets
+        .targets()
+        .get(&VirtualTargetPath::new(step.to_string()).unwrap())
+        .unwrap()
+        .hashes()
+        .get(&HashAlgorithm::Sha256)
+        .unwrap();
+
+    let target_str = if consistent_snapshot {
+        format!("{}.{}", hash, step.to_string())
+    } else {
+        step.to_string()
+    };
+    let target_path = TargetPath::new(target_str).unwrap();
+    repo.store_target(&mut &*target_data, &target_path)
+        .await
+        .unwrap();
+
+    let version_prefix = if consistent_snapshot {
+        MetadataVersion::Number(version)
+    } else {
+        MetadataVersion::None
+    };
+
+    repo.store_metadata(
+        &targets_path,
+        &version_prefix,
+        &mut signed_targets.to_raw().unwrap().as_bytes(),
+    )
+    .await
+    .unwrap();
+
+    let snapshot_path = MetadataPath::from_role(&Role::Snapshot);
+    let snapshot = SnapshotMetadataBuilder::new()
+        .expires(expiration)
+        .version(version)
+        .insert_metadata(&signed_targets, &[HashAlgorithm::Sha256])
+        .unwrap()
+        .signed::<JsonPretty>(keys.get("snapshot").unwrap())
+        .unwrap();
+
+    repo.store_metadata(
+        &snapshot_path,
+        &version_prefix,
+        &mut snapshot.to_raw().unwrap().as_bytes(),
+    )
+    .await
+    .unwrap();
+
+    let timestamp_path = MetadataPath::from_role(&Role::Timestamp);
+    let timestamp = TimestampMetadataBuilder::from_snapshot(&snapshot, &[HashAlgorithm::Sha256])
+        .unwrap()
+        .expires(expiration)
+        .version(version)
+        .signed::<JsonPretty>(keys.get("timestamp").unwrap())
+        .unwrap();
+
+    // Timestamp doesn't require a version prefix even in consistent_snapshot.
+    repo.store_metadata(
+        &timestamp_path,
+        &MetadataVersion::None,
+        &mut timestamp.to_raw().unwrap().as_bytes(),
+    )
+    .await
+    .unwrap();
+}
+
+pub async fn generate_repos(
+    keys_path: &Path,
+    dir: &Path,
+    consistent_snapshot: bool,
+) -> tuf::Result<()> {
+    // Create initial repo.
+    let json_keys = init_json_keys(keys_path);
+    let mut keys = init_role_keys(&json_keys);
+    let dir0 = Path::new(dir).join("0");
+    let repo = FileSystemRepositoryBuilder::new(dir0)
+        .metadata_prefix(Path::new("repository"))
+        .targets_prefix(Path::new("repository").join("targets"))
+        .build()?;
+
+    update_root(&repo, &keys, None, 1, consistent_snapshot).await;
+    add_target(&repo, &keys, 0, consistent_snapshot).await;
+
+    let mut i: u8 = 1;
+    let rotations = vec![
```

Comment:
A comment here would be helpful

Reply:
Added some docs. Please let me know if that helped.

Reply:
Yup, thanks.